### PR TITLE
vol2heightfield  and heightfield2vol

### PR DESCRIPTION
**Pull request by kerautret**
Two simple tools to convert vol file and heightfield image :
For instance you can obtain an height map from a given direction:
 ./converters/vol2heightfield -i ../../DGtal/examples/samples/lobster.vol -m 60 -M 500  --nx 0 --ny 0.7 --nz -1 -x 150 -y 0 -z 150 --width 300 --height 300 --heightFieldMaxScan 350  -o tmp.pgm 
tmp
and from an height map you can obtain a vol file:
./converters/heightfield2vol -i ${DGtal}/examples/samples/church.pgm -o volResu.vol -s 0.3 -z 50 
capture d ecran 2015-03-18 a 11 22 03

### CODE REVIEW

File: converters/heightfield2vol.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  typedef DGtal::ConstImageAdapter<Image3D, Z2i::Domain, DGtal::functors::Point2DEmbedderIn3D<DGtal::Z3i::Domain>,
+                                   Image3D::Value,  DGtal::functors::Identity >  ImageAdapterExtractor;
+
+  // parse command line ----------------------------------------------
+  po::options_description general_opt("Allowed options are: ");
+  general_opt.add_options()
+    ("help,h", "display this message")
+    ("input,i", po::value<std::string>(), "heightfield file." )
+    ("output,o", po::value<std::string>(), "volumetric file ") 
+    ("scale,s", po::value<double>()->default_value(1.0), "set the scale of the maximal level. (default 1.0)")
```
Comment (dcoeurjo) [inline converters/heightfield2vol.cpp]: set the scale factor on height values ?

File: converters/heightfield2vol.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    ("volZ,z", po::value<double>()->default_value(255), "set the Z max value of domain.")
```
Comment (dcoeurjo) [inline converters/heightfield2vol.cpp]: double -> uint32_t ?

Comment (dcoeurjo) [inline converters/heightfield2vol.cpp]: (if it is used to construct the domain)

File: converters/heightfield2vol.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  // parse command line ----------------------------------------------
+  po::options_description general_opt("Allowed options are: ");
+  general_opt.add_options()
+    ("help,h", "display this message")
+    ("input,i", po::value<std::string>(), "heightfield file." )
+    ("output,o", po::value<std::string>(), "volumetric file ") 
+    ("scale,s", po::value<double>()->default_value(1.0), "set the scale of the maximal level. (default 1.0)")
+    ("volZ,z", po::value<double>()->default_value(255), "set the Z max value of domain.")    
+    ("foregroundValue,f", po::value<unsigned int>()->default_value(128), "specify the foreground value of the resulting voxel." )
+    ("backgroundValue,b", po::value<unsigned int>()->default_value(0), "change the default background (black with the last filled intensity).");
```
Comment (dcoeurjo) [inline converters/heightfield2vol.cpp]: is it the background value of the resulting vol file ?

File: converters/vol2heightfield.cpp [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+  // parse command line ----------------------------------------------
+  po::options_description general_opt("Allowed options are: ");
+  general_opt.add_options()
+    ("help,h", "display this message")
+    ("input,i", po::value<std::string>(), "volumetric file (.vol) " )
+    ("output,o", po::value<std::string>(), "sequence of discrete point file (.sdp) ") 
+    ("thresholdMin,m", po::value<int>()->default_value(128), "min threshold (default 128)" )
+    ("thresholdMax,M", po::value<int>()->default_value(255), "max threshold (default 255)" )
+    ("nx", po::value<double>()->default_value(0), "choose x comp of the normal vector." )
```
Comment (dcoeurjo) [inline converters/vol2heightfield.cpp]: set the x component of the projection direction
(and similarly for the other options?)

File: converters/vol2heightfield.cpp [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    ("ny", po::value<double>()->default_value(0), "choose y comp of the normal vector." )
+    ("nz", po::value<double>()->default_value(1), "choose z  comp of the normal vector." )
+    ("centerX,x", po::value<unsigned int>()->default_value(0), "choose x center of the projected first image." )
```
Comment (dcoeurjo) [inline converters/vol2heightfield.cpp]: why first?

**Comment by dcoeurjo** [comment]
nice tools.. I love the HF to vol to generate huge vol files.

File: converters/heightfield2vol.cpp (hunk 1 above)
Comment (kerautret) [inline converters/heightfield2vol.cpp]: yes better, thanks

File: converters/heightfield2vol.cpp (hunk 2 above)
Comment (kerautret) [inline converters/heightfield2vol.cpp]: yes thank I forget to use it after

File: converters/heightfield2vol.cpp (hunk 3 above)
Comment (kerautret) [inline converters/heightfield2vol.cpp]: yes (old copy/parst) ;)

File: converters/vol2heightfield.cpp (hunk 4 above)
Comment (kerautret) [inline converters/vol2heightfield.cpp]: thanks for the better suggestion

File: converters/vol2heightfield.cpp (hunk 5 above)
Comment (kerautret) [inline converters/vol2heightfield.cpp]: related to the algorithm but agree not really needed for the user, i remove it

**Comment by kerautret** [comment]
Thank for the review, that corrected ;)

**Comment by dcoeurjo** [comment]
great merging

**Pull request merged**